### ISSUE TITLE
ESLint 9 support

### ISSUE THREAD

**Issue by ai** [sveltejs/eslint-plugin-svelte#726]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.0.0

### What version of `eslint-plugin-svelte` are you using?

2.35.1

### What did you do?

I have flat config:

<details>
<summary>Configuration</summary>

```
import svelte from 'eslint-plugin-svelte'
import svelteParser from 'svelte-eslint-parser'

export default [
  {
    languageOptions: {
      parserOptions: {
        extraFileExtensions: ['.svelte'],
        project: true
      },
      sourceType: 'module'
    }
  },
  {
    files: ['**/*.svelte'],
    languageOptions: {
      parser: svelteParser
    },
    plugins: {
      svelte
    },
    processor: 'svelte/svelte'
  }
]
```
</details>

```svelte
<script lang="ts">
  let href: any
</script>

<a {href}>Link</a>
```

### What did you expect to happen?

No errors

### What actually happened?

Error:

```
Oops! Something went wrong! :(

ESLint: 9.0.0

Error: This method cannot be used with flat config. Add your entries directly into the config array.
    at assertEslintrcConfig (/home/ai/Dev/logux-eslint-config/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/linter/linter.js:1286:15)
    at Linter.getRules (/home/ai/Dev/logux-eslint-config/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/linter/linter.js:2215:9)
    at getCoreRule (/home/ai/Dev/logux-eslint-config/node_modules/.pnpm/eslint-plugin-svelte@2.35.1_eslint@9.0.0_svelte@4.2.12/node_modules/eslint-plugin-svelte/lib/utils/eslint-core.js:51:47)
    at Object.<anonymous> (/home/ai/Dev/logux-eslint-config/node_modules/.pnpm/eslint-plugin-svelte@2.35.1_eslint@9.0.0_svelte@4.2.12/node_modules/eslint-plugin-svelte/lib/rules/no-inner-declarations.js:5:48)
    at Module._compile (node:internal/modules/cjs/loader:1368:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1426:10)
    at Module.load (node:internal/modules/cjs/loader:1205:32)
    at Module._load (node:internal/modules/cjs/loader:1021:12)
    at Module.require (node:internal/modules/cjs/loader:1230:19)
    at require (node:internal/modules/helpers:179:18)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/logux/eslint-config/blob/eslint9-svelte/svelte.js

```
pnpm install
pnpm eslint demo/index.{js,ts,svelte} --config svelte.js
```

**Comment by ai** [comment]
Fixed. Thanks.

What do you think of keeping this issue open until it will be released to avoid creating duplicates?

**Comment by ota-meshi** [comment]
Yeah. We will keep this issue open until we release v2.36.0.
